Essentials/src/net/ess3/api/events/TransactionEvent.java

```diff
@@ -0,0 +1,53 @@
+package net.ess3.api.events;
+
+import com.earth2me.essentials.CommandSource;
+import net.ess3.api.IUser;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+import java.math.BigDecimal;
+
+public class TransactionEvent extends Event {
+    private static final HandlerList handlers = new HandlerList();
+
+    private final CommandSource requester;
+    private final IUser target;
+    private final BigDecimal amount;
+
+    public TransactionEvent(CommandSource requester, IUser target, BigDecimal amount) {
+        super();
```

Comment:
Should probably check the current thread here, in case transactions occur async.

Reply:
Not sure what you mean, doesn't the event get raised after the transaction is over?